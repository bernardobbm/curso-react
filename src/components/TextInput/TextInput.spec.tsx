import { describe, expect, it, vitest } from 'vitest';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import { TextInput } from '.';

const fn = vitest.fn();

describe('<TextInput />', () => {
	it('should have a value on searchValue', () => {
		render(
			<TextInput
				searchValue="testing"
				handleInputChange={fn}
			/>
		);

		const inputText = screen.getByPlaceholderText(/type your search/i) as HTMLInputElement;
		expect(inputText.value).toBe('testing');
	});

	it('should call handleChange function on each key pressed', async () => {
		render(
			<TextInput
				searchValue="this first value"
				handleInputChange={fn}
			/>
		);

		const value = 'this value';

		const inputText = screen.getByPlaceholderText(/type your search/i) as HTMLInputElement;

		await userEvent.type(inputText, value);

		expect(inputText.value).toBe('this first value');
		expect(fn).toHaveBeenCalledTimes(value.length);
	});

	it('should match snapshot', () => {
		const { container } = render(
			<TextInput
				searchValue=""
				handleInputChange={fn}
			/>
		);

		expect(container.firstChild).toMatchSnapshot();
	});
});
