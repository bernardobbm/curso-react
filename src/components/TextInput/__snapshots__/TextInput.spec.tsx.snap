// Vitest Snapshot v1

exports[`<TextInput /> > should match snapshot 1`] = `
<input
  class="text-input"
  placeholder="Type your search"
  type="search"
  value=""
/>
`;
